src/Interactive/EditorFeatures/Core/Extensibility/Interactive/NuGetPackageResolverImpl.cs

```diff
@@ -0,0 +1,204 @@
+        }
+
+        /// <summary>
+        /// Syntax is "id/version", matching references in project.lock.json.
+        /// </summary>
+        internal static bool ParsePackageReference(string reference, out string name, out string version)
+        {
+            var parts = reference.Split('/');
+            int n = reference.Length;
+            if ((parts.Length == 2) &&
+                (parts[0].Length > 0) &&
+                (parts[1].Length > 0))
+            {
+                name = parts[0];
+                version = parts[1];
+                return true;
+            }
+            name = null;
+            version = null;
+            return false;
+        }
+
+        /// <summary>
+        /// Generate a project.json file with the packages as "dependencies".
+        /// </summary>
+        internal static void WriteProjectJson(TextWriter writer, string packageName, string packageVersion)
+        {
+            using (var jsonWriter = new JsonTextWriter(writer))
+            {
+                jsonWriter.Formatting = Newtonsoft.Json.Formatting.Indented;
+                jsonWriter.WriteStartObject();
+                // "dependencies" : { "packageName" : "packageVersion" }
+                jsonWriter.WritePropertyName("dependencies");
+                jsonWriter.WriteStartObject();
+                jsonWriter.WritePropertyName(packageName, escape: true);
+                jsonWriter.WriteValue(packageVersion);
+                jsonWriter.WriteEndObject();
+                // "frameworks" : { "net46" : {} }
+                jsonWriter.WritePropertyName("frameworks");
+                jsonWriter.WriteStartObject();
+                jsonWriter.WritePropertyName(ProjectJsonFramework, escape: true);
+                jsonWriter.WriteStartObject();
+                jsonWriter.WriteEndObject();
+                jsonWriter.WriteEndObject();
+                jsonWriter.WriteEndObject();
+            }
+        }
+
+        /// <summary>
+        /// Read the references from the project.lock.json file.
+        /// </summary>
+        internal static ImmutableArray<string> ReadProjectLockJson(string packagesDirectory, TextReader reader)
+        {
+            JObject obj;
+            using (var jsonReader = new JsonTextReader(reader))
+            {
+                obj = JObject.Load(jsonReader);
+            }
+            var builder = ArrayBuilder<string>.GetInstance();
+            var targets = (JObject)GetPropertyValue(obj, "targets");
+            foreach (var target in targets)
+            {
+                if (target.Key == ProjectLockJsonFramework)
+                {
+                    foreach (var package in (JObject)target.Value)
+                    {
+                        var packageRoot = PathUtilities.CombineAbsoluteAndRelativePaths(packagesDirectory, package.Key);
+                        var runtime = (JObject)GetPropertyValue((JObject)package.Value, "runtime");
```

Comment:
FYI the runtime section for net46 may not be correct for all cases...we really should have you generating a runtime group as well.